src/System.Web.OData/OData/Formatter/EdmLibHelpers.cs

```diff
@@ -825,12 +831,15 @@ public static IEnumerable<IEdmStructuralProperty> GetConcurrencyProperties(this
                 }
             }
 
-            if (_concurrencyProperties == null)
+            if (concurrencyProperties == null)
             {
-                _concurrencyProperties = new ConcurrentDictionary<IEdmNavigationSource, IEnumerable<IEdmStructuralProperty>>();
+                concurrencyProperties = new ConcurrentDictionary<IEdmNavigationSource, IEnumerable<IEdmStructuralProperty>>();
             }
```

Comment:
You shouldn't need Lines 834-837 since you already made sure concurrencyProperties was non-null in 792.